modules/javafx.controls/src/test/java/test/javafx/scene/control/ListCellTest.java

```diff
@@ -852,6 +852,28 @@ public void testEditCancelEventAfterRemoveEditingItem() {
         assertEquals("editing location of cancel event", editingIndex, events.get(0).getIndex());
     }
 
+    @Test
+    public void testStartEditOffRangeMustNotFireStartEdit() {
+        list.setEditable(true);
+        cell.updateListView(list);
+        cell.updateIndex(list.getItems().size());
+        List<EditEvent> events = new ArrayList<>();
+        list.addEventHandler(ListView.editStartEvent(), events::add);
+        cell.startEdit();
+        assertFalse("sanity: off-range cell must not be editing", cell.isEditing());
+        assertEquals("must not fire editStart", 0, events.size());
+    }
+
+    @Test
+    public void testStartEditOffRangeMustNotUpdateEditingLocation() {
+        list.setEditable(true);
+        cell.updateListView(list);
+        cell.updateIndex(list.getItems().size());
+        cell.startEdit();
+        assertFalse("sanity: off-range cell must not be editing", cell.isEditing());
+        assertEquals("list editing location must not be updated", - 1, list.getEditingIndex());
```

Comment:
`-1` would look better here (without the space inbetween) given the fact this is not a mathematical context